### DISCUSSION TITLE
How to dynamically execute code and capture it's output?

### DISCUSSION THREAD

**Discussion by Kossak-esky** [xonsh/xonsh#5332] [Q&A]
How to dynamically execute code and capture it's output?

Example script:
```xsh
#!/usr/bin/env xonsh

cmd = '''date +%H%M'''
p = evalx(cmd)
print('stdout: ', p.stdout)
```

I'd like to be able to capture output of cmd, which in this case can be for example `1244`, but above code prints `None` as stdout. I know I can just use python subprocess, but maybe there's some shorter way to do it using some xonsh commands (`$()` maybe)?
Thank you.

**Comment by Spitfire1900** [comment]
What is the value of `$XONSH_HISTORY_BACKEND`?

**Comment by Kossak-esky** [comment]
It's `json`:

```
$ echo $XONSH_HISTORY_BACKEND
json
```

**Comment by Spitfire1900** [comment]
Not quite an answer, but 

```
@ type(evalx(_cmd))
1146
xonsh.procs.pipelines.HiddenCommandPipeline
```

Per https://xon.sh/subproc_types.html I don't think HiddenCommandPipeline captures the output as something you can store for later use.

**Comment by Kossak-esky** [comment]
I noticed that `p` is `HiddenCommandPipeline` before, but you got me thinking now and I got:
```
cmd = '''date'''
p = !(@(cmd))
print(f'Output is: {p.out}')
```

which works as I wanted to and is shorter than subprocess version. Thank you :)
